Test/CoreSDK.Test/Shared/Extensibility/Implementation/TaskTimerInternalTest.cs

```diff
@@ -203,6 +210,34 @@ public void AbortsPreviousAction()
                     Assert.False(actionInvoked);
                 });
             }
+
+            [TestMethod]
+            public void CancelAbortsOnlyLastTask()
+            {
+                var timer = new TaskTimerInternal { Delay = TimeSpan.FromMilliseconds(1) };
+
+                int invokationCount = 0;
+                var firstActionInvoked = new ManualResetEventSlim();
+                var secondActionInvoked = new ManualResetEventSlim();
+                timer.Start(() => Task.Factory.StartNew(
+                    () =>
+                    {
+                        Interlocked.Increment(ref invokationCount);
+                        firstActionInvoked.Set();
+                    }));
+                timer.Start(
+                    () => Task.Factory.StartNew(
+                        () =>
+                        {
+                            Interlocked.Increment(ref invokationCount);
+                            secondActionInvoked.Set();
+                        }));
+
+                timer.Cancel();
```

Comment:
1 ms may not be sufficient between the `Start` and `Cancel`. You are waiting 50 milliseconds so maybe change timer to 25 milliseconds?